### PR TITLE
PR: Show size of objects correctly, if they implement len()

### LINKED ISSUE #245: Display size of string in variable editor
Currently the variable explorer displays all strings with size `1` - I think it would be more intuitive to show the length of the string.

It would simply requiring to change the size here to include `str`

https://github.com/spyder-ide/spyder-kernels/blob/87d89ad94becb51f4eea1bea643334fd02440d6c/spyder_kernels/utils/nsview.py#L124

### PR DESCRIPTION
**Pull request by skjerns**
#245  I suggested to show the string size correctly and not display `1`.

I thought I even extend that to all objects implementing `len()`. I think if the object can tell us how large it is, we can display this self-reported size.

Fixes #245.

### CODE REVIEW

**Review by dalthviz — COMMENTED** [review]
Thanks @skjerns ! I think we could add a test for this, could you please add one? (you can add it [here](https://github.com/spyder-ide/spyder-kernels/blob/1.x/spyder_kernels/utils/tests/test_nsview.py)), otherwise this LGTM :+1:

**Comment by skjerns** [comment]
I've added tests. Additionally, I extended the functionality such that it will show either `x.shape`, `x.size` or `len(x)`. I think this should cover 98% of objects that have a length.

File: spyder_kernels/utils/tests/test_nsview.py [hunk 1]
```diff
@@ -18,12 +18,13 @@
 import pandas as pd
 import pytest
 import xarray as xr
+import PIL
```
Comment (ccordoba12) [inline spyder_kernels/utils/tests/test_nsview.py:21]: Please add `pillow` to our requirements for our tests in our `setup.py` and `requirements/tests.txt`.

**Review by ccordoba12 — COMMENTED** [review]
Hey @skjerns, thanks for the new test. Unfortunately you merged `master` with this branch, which was incorrect because this PR needs to be against our `1.x` branch.

Could you fix that?

**Comment by skjerns** [comment]
> Hey @skjerns, thanks for the new test. Unfortunately you merged `master` with this branch, which was incorrect because this PR needs to be against our `1.x` branch.
> 
> Could you fix that?

Oh no, not again 😅 I'll try to fix it...

**Comment by ccordoba12** [comment]
> It might be preferable to make a new branch and open a new PR for this. What do you think?

Totally agree, if that's not much to ask.

**Comment by skjerns** [comment]
no problem! much easier than rebasing :) continued at #250

**Pull request closed without merging**